components/domains/registerV2.tsx

```diff
@@ -87,6 +90,20 @@ const RegisterV2: FunctionComponent<RegisterV2Props> = ({ domain }) => {
   );
   const { addTransaction } = useTransactionManager();
 
+  // on first load, we generate a salt
+  useEffect(() => {
+    setSalt(generateSalt());
+  }, []);
```

Comment:
Why can't you generate the salt on the second useEffect because we only need it here ?

Reply:
You need it both to compute the hash and to send it to the server for verification